src/lib/core/GroupId.h

```diff
@@ -0,0 +1,51 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <cstdint>
+
+namespace chip {
+
+// TODO: Consider making this a class and the various utility methods static
+// methods.
+using GroupId = uint16_t;
+
+constexpr GroupId kUndefinedGroupId = 0;
+
+constexpr GroupId kMinUniversalGroupId = 0x8000;
+constexpr GroupId kMaxUniversalGroupId = 0xFFFF;
```

Comment:
Would it be preferable to explicitly define the reserved range (0x8000 - 0xFFFC), all proxies (0xFFFD), all sleep devices (0xFFFE) and all nodes (0xFFFF) codes?

Reply:
Sure, added